R/check.R

```diff
@@ -137,46 +136,36 @@ solve_ip.min_isolated_deps_installation_proposal <- function(ip) { # nolint
 
   cli_pb_init("min_isolated", total = nrow(deps))
 
-  deps_res <- lapply(
-    seq_len(nrow(deps)),
-    function(i) {
-      i_pkg <- deps[i, "package"]
+  deps_res <- lapply(seq_len(nrow(deps)), function(i) {
+    i_pkg <- deps[i, "package"]
 
-      cli_pb_update(package = i_pkg, n = 4L)
+    cli_pb_update(package = i_pkg, n = 4L)
 
-      if (i_pkg %in% base_pkgs()) {
-        return(NULL)
-      }
+    if (i_pkg %in% base_pkgs()) return(NULL)
 
-      i_op <- deps[i, "op"]
-      i_op_ver <- deps[i, "version"]
+    result <- resolve_ppm_snapshot(
+      deps[i, "ref"],
+      deps[i, "op"],
+      deps[i, "version"]
+    )
 
-      i_ref_str <- deps[i, "ref"]
-      i_ref <- pkgdepends::parse_pkg_ref(i_ref_str)
-
-      i_ref_minver <- get_ref_min_incl_cran(i_ref, i_op, i_op_ver)
-
-      i_release_date <- get_release_date(i_ref_minver)
+    # Remove duplicate entries of current package
+    result[result$ref != deps[i, "ref"] | !duplicated(result$ref), ]
+  })
 
-      if (is.na(i_release_date)) {
-        ppm_repo <- file.path(pkgcache::ppm_repo_url(), "latest")
-      } else {
-        ppm_repo <- parse_ppm_url(get_ppm_snapshot_by_date(i_release_date))
-      }
+  new_res <- do.call(rbind, deps_res)
+  new_res <- new_res[!duplicated(new_res[, c("ref", "package", "version")]), ]
 
-      i_pkg_deps <- pkgdepends::new_pkg_deps(
-        if (inherits(i_ref_minver, "remote_ref_github")) i_ref_minver$ref else i_ref$ref,
-        config = list(dependencies = "hard", cran_mirror = ppm_repo)
-      )
-      suppressMessages(i_pkg_deps$resolve())
-      i_res <- i_pkg_deps$get_resolution()
-      i_res$direct <- i_res$directpkg <- FALSE
-      i_res
-    }
-  )
+  # Keep primary dependencies versions (in case primary dependency is
+  #  resolved in other dependencies)
+  new_res <- new_res[
+    !(new_res$package %in% deps$package) |
+      new_res$package == new_res$parent
+    ,
+  ]
```

Comment:
as discussed in our chat conversation, this highlighted part needs to go away
otherwise is a good change

Reply:
Done! :heavy_check_mark: